### PR TITLE
Implement tokens sub-commands.

### PR DESCRIPTION
**Pull request by andrewsomething**
Opening as a draft until https://github.com/digitalocean/godo/pull/591 is released.

### CODE REVIEW

File: commands/tokens.go [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+	- The name of the token
+	- The scopes applied to the token
+	- The timestamp for when the token will expire
+	- The date that it was last used
+	- The timestamp for when it was created
+	`)
+
+	CmdBuilder(cmd, RunTokenGet, "get <id|name>",
+		"Retrieve information about a DigitalOcean API token",
+		heredoc.Docf(`Display the details for a specific DigitalOcean API tokens on your account. This includes:
```
Comment (bentranter) [inline commands/tokens.go:45]: Small typo
```suggestion
		heredoc.Docf(`Display the details for a specific DigitalOcean API token on your account. This includes:
```

File: commands/tokens.go [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+	CmdBuilder(cmd, RunTokenList, "list",
+		"List your DigitalOcean API tokens",
+		heredoc.Docf(`Lists the following details for DigitalOcean API tokens on your account:
+
+		%s`, tokenDetails),
+		Writer, aliasOpt("ls"), displayerType(&displayers.Tokens{}))
+
+	tokensCreateDesc := heredoc.Docf(`Create a new DigitalOcean API token with granular scopes.
+
+		You can find all of the scopes that are available to be applied to a tokens using the
```
Comment (bentranter) [inline commands/tokens.go:59]:
```suggestion
		You can find all of the scopes that are available to be applied to a token using the
```

File: commands/tokens.go [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+		%s sub-command. For example, to create a token that can be used to create and manage
+		Droplets, use:
+
+		    doctl token create droplet-service-token --scopes droplet:create,droplet:read,droplet:delete,droplet:operate
+
+		By default, tokens will expire in 30 days (2592000 seconds) by default. The maximum
```
Comment (bentranter) [inline commands/tokens.go:65]: The second "by default" is redundant.
```suggestion
		By default, tokens will expire in 30 days (2592000 seconds). The maximum
```

File: commands/tokens.go [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+	AddStringFlag(createTokenCmd, doctl.ArgTokenExpiresIn, "", "",
+		`Duration (e.g. 24h, 60m, or 3600s) from now until the token will expire. Valid units are: "h", "m", "s".`)
+	AddStringSliceFlag(createTokenCmd, doctl.ArgTokenScopes, "", nil,
+		`Scopes to assign to the to token (e.g. droplet:read)`, requiredOpt())
+
+	updateTokenCmd := CmdBuilder(cmd, RunTokenUpdate, "update <id|name>",
+		"Update an existing DigitalOcean API token",
+		heredoc.Docf(`Update the name or scopes for an existing DigitalOcean API token.
+
+		You can find all of the scopes that are available to be applied to a tokens using the
```
Comment (bentranter) [inline commands/tokens.go:85]:
```suggestion
		You can find all of the scopes that are available to be applied to a token using the
```

File: commands/tokens.go [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+		or destroy one.
+
+		The list of scopes can be filtered by namespace using the %s flag.`, "`--namespace`"),
+		Writer, aliasOpt("ls"), displayerType(&displayers.Tokens{}))
+	AddStringFlag(listScopesCmd, doctl.ArgTokenScopeNamespace, "", "", "Filter scopes by their namespace")
+
+	return cmd
+}
+
+// RunTokenGet returns a single of token.
```
Comment (bentranter) [inline commands/tokens.go:116]:
```suggestion
// RunTokenGet returns a single API token.
```

**Review by bentranter — APPROVED** [review]
A few typos, but other than that LGTM

**Review by bentranter — APPROVED** [review]

**Comment by danaelhe** [comment]
Looks great! Very natural command line look and feel. It seems that the "token" command might be missing from the "available commands" list? Is that intentional for now?

```
Available Commands:
  1-click         Display commands that pertain to 1-click applications
  account         Display commands that retrieve account details
  apps            Display commands for working with apps
  auth            Display commands for authenticating doctl with an account
  balance         Display commands for retrieving your account balance
  billing-history Display commands for retrieving your billing history
  completion      Generate the autocompletion script for the specified shell
  compute         Display commands that manage infrastructure
  databases       Display commands that manage databases
  help            Help about any command
  invoice         Display commands for retrieving invoices for your account
  kubernetes      Displays commands to manage Kubernetes clusters and configurations
  monitoring      [Beta] Display commands to manage monitoring
  projects        Manage projects and assign resources to them
  registry        Display commands for working with container registries
  serverless      Develop, test, and deploy serverless functions
  version         Show the current version
  vpcs            Display commands that manage VPCs
```

**Comment by andrewsomething** [comment]
Yes, that is intentional as this is a very limited release. 

https://github.com/digitalocean/doctl/blob/dcca9fdfbe620a1ac70c945b31f48ceb60ffc2b5/commands/tokens.go#L30

**Review by scotchneat — APPROVED** [review]
Great stuff ❤️

**Pull request merged**